### PR TITLE
add process script and update typedoc config

### PR DESCRIPTION
**Pull request by jacoblogan**
#### Description of changes
As part of the api docs migration typedoc will be exported as a json file and then flattened to be used by the docs repo to generate the api docs pages.

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Unit Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

#### Checklist for repo maintainers

- [ ] Verify E2E tests for existing workflows are working as expected or add E2E tests for newly added workflows
- [ ] New source file paths included in this PR have been added to CODEOWNERS, if appropriate

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: scripts/process-typedoc.mjs [hunk 1]
```diff
@@ -0,0 +1,77 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import references from '../docs/reference.json' assert { type: 'json' };
+import { writeFileSync } from 'fs';
+
+// build flat object for easier faster lookups
+const flatReferences = {};
+
+const recursivelyPopulateFlatObject = referenceObject => {
```
Comment (jimblanc) [inline scripts/process-typedoc.mjs:9]: Some inline documentation on the various steps in this script would be really helpful for future maintainers and those not familiar with the API docs, for example: https://github.com/aws-amplify/amplify-js/blob/main/scripts/set-preid-versions.sh

Also could you please document in the script which versions of TypeDoc this is compatible with?

File: scripts/process-typedoc.mjs [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+	return cat;
+});
+
+try {
+	writeFileSync(
+		'docs/parsedJson.json',
+		JSON.stringify(flatReferences, null, 2),
+		'utf8',
+	);
+	console.log('Data successfully saved to disk');
```
Comment (jimblanc) [inline scripts/process-typedoc.mjs:74]: Could we make these messages a little more descriptive to help debugging if something goes wrong, e.g.:

```suggestion
	console.log('Successfully saved parsed API information');
```

File: scripts/process-typedoc.mjs [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+} catch (error) {
+	console.log('An error has occurred ', error);
```
Comment (jimblanc) [inline scripts/process-typedoc.mjs:76]: Sanity check: If something failed in this script, will the release process be disrupted? Or does it depend on the eventual workflow implementation?

File: scripts/process-typedoc.mjs (hunk 1 above)
Comment (jacoblogan) [reply scripts/process-typedoc.mjs:27]: Thanks inline documentation has been added to the script

File: scripts/process-typedoc.mjs (hunk 3 above)
Comment (jacoblogan) [reply scripts/process-typedoc.mjs:98]: You are correct the error here is swallowed up and will not disrupt the release process.  The workflow implementation will look for changes to the typedoc output, but as discussed earlier we don't want this process to block the release.

File: scripts/process-typedoc.mjs
Comment (jimblanc) [inline scripts/process-typedoc.mjs:1]: File missing copyright header at top

**Pull request merged**